### PR TITLE
Added CLI acceptance tests for `occ` encryption command

### LINKED ISSUE #39189: add cli tests for encryption command
from https://github.com/owncloud/core/issues/33051

add cli tests for encryption command

### PR DESCRIPTION
**Pull request by kiranparajuli589**
## Description
- added `cli` acceptance tests for `occ` encryption command

## Related Issue

- Fixes #39189 

## How Has This Been Tested?

- 🤖 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>cliEncryption-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/32340/122/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>cliEncryption-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/32346/122/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>cliEncryption-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/32350/122/1

**Comment by kiranparajuli589** [comment]
@phil-davis 
Screenshot from 2021-09-15 11-56-11

for the `occ` commands run as `system` commands, this error is thrown (I'm facing this locally too) the config directory and all its children has `www-data` as owner and has write access too...but the test is executing as `admin`.

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>cliEncryption-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/32477/122/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>cliEncryption-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/32478/5/1

**Comment by kiranparajuli589** [comment]
skipped https://github.com/owncloud/encryption/issues/303

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>cliEncryption-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/32533/122/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>cliEncryption-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/32588/122/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>cliEncryption-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/32592/122/1

File: tests/acceptance/features/bootstrap/OccContext.php [hunk 1]
```diff
@@ -711,6 +721,17 @@ public function theAdministratorRunsUpgradeRoutinesOnLocalServerUsingTheOccComma
 		}
 	}
 
+	/**
+	 * @When the administrator runs encryption decrypt all using the occ command
+	 *
+	 * @return void
+	 */
+	public function theAdministratorRunsEncryptionDecryptAllUsingTheOccCommand() {
```
Comment (SwikritiT) [inline tests/acceptance/features/bootstrap/OccContext.php:729]:
```suggestion
	public function theAdministratorRunsEncryptionDecryptAllUsingTheOccCommand():void {
```

File: tests/acceptance/features/cliEncryption/encryption.feature [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    When the administrator runs encryption decrypt all using the occ command
+    And the administrator invokes occ command "encryption:disable"
+    Then the command should have been successful
+    And the command output should contain the text "Cleaned up config"
+    And the command output should contain the text "Encryption is already disabled"
+    When the administrator invokes occ command "encryption:status"
+    Then the command should have been successful
+    And the command output should contain the text "enabled: false"
+
+
```
Comment (saw-jan) [inline tests/acceptance/features/cliEncryption/encryption.feature:65]: extra new lines
```suggestion
```

File: tests/acceptance/features/cliEncryption/encryption.feature [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    And the command output should contain the text "Current key storage root:  default storage location (data/)"
+
+
+  Scenario: move encryption keys to a different folder
+    When the administrator invokes occ command "encryption:change-key-storage-root owncloud-keys"
+    Then the command should have been successful
+    When the administrator invokes occ command "encryption:show-key-storage-root"
+    And the command output should contain the text "Current key storage root:  owncloud-keys"
+
+  Scenario: data file contents should be encrypted
```
Comment (saw-jan) [inline tests/acceptance/features/cliEncryption/encryption.feature:37]: two lines between scenarios like above and below
```suggestion
    And the command output should contain the text "Current key storage root:  owncloud-keys"


  Scenario: data file contents should be encrypted
```

File: tests/acceptance/features/bootstrap/OccContext.php [hunk 4]
```diff
@@ -711,6 +721,17 @@ public function theAdministratorRunsUpgradeRoutinesOnLocalServerUsingTheOccComma
...
+		\system("echo 'admin' | sudo -S www-data ./occ encryption:decrypt-all -c yes", $status);
+		\var_dump($status);
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/OccContext.php:731]: Debug line?
Or is this useful to commit?

File: tests/acceptance/features/cliEncryption/encryption.feature [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    When the administrator invokes occ command "encryption:show-key-storage-root"
+    Then the command should have been successful
+    And the command output should contain the text "Current key storage root:  default storage location (data/)"
+
+
+  Scenario: move encryption keys to a different folder
+    When the administrator invokes occ command "encryption:change-key-storage-root owncloud-keys"
+    Then the command should have been successful
+    When the administrator invokes occ command "encryption:show-key-storage-root"
+    And the command output should contain the text "Current key storage root:  owncloud-keys"
```
Comment (phil-davis) [inline tests/acceptance/features/cliEncryption/encryption.feature:35]:
```suggestion
    Then the command output should contain the text "Current key storage root:  owncloud-keys"
```

In  this case it's OK to have `Then` twice - the first one is just a double-check of the first occ command.

File: tests/acceptance/features/cliEncryption/encryption.feature [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+  Scenario: data file contents should be encrypted
+    Given user "Alice" has been created with default attributes and without skeleton files
+    When user "Alice" uploads file with content "file to upload" to "/fileToUpload.txt" using the WebDAV API
+    Then file "fileToUpload.txt" of user "Alice" should be encrypted
+
+
+  Scenario: downloaded content of an uploaded should not be encrypted
+    Given user "brand-new-user" has been created with default attributes and without skeleton files
+    When user "brand-new-user" uploads file with content "uploaded content" to "fileToUpload.txt" using the WebDAV API
+    And the content of file "fileToUpload.txt" for user "brand-new-user" should be "uploaded content"
```
Comment (phil-davis) [inline tests/acceptance/features/cliEncryption/encryption.feature:46]:
```suggestion
    Then the content of file "fileToUpload.txt" for user "brand-new-user" should be "uploaded content"
```

File: tests/acceptance/features/cliEncryption/encryption.feature [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+    When the administrator invokes occ command "encryption:show-key-storage-root"
+    And the command output should contain the text "Current key storage root:  owncloud-keys"
+
+  Scenario: data file contents should be encrypted
+    Given user "Alice" has been created with default attributes and without skeleton files
+    When user "Alice" uploads file with content "file to upload" to "/fileToUpload.txt" using the WebDAV API
+    Then file "fileToUpload.txt" of user "Alice" should be encrypted
+
+
+  Scenario: downloaded content of an uploaded should not be encrypted
```
Comment (phil-davis) [inline tests/acceptance/features/cliEncryption/encryption.feature:43]:
```suggestion
  Scenario: downloaded content of an uploaded file should not be encrypted
```

File: tests/acceptance/features/cliEncryption/encryption.feature [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+    Given user "brand-new-user" has been created with default attributes and without skeleton files
+    When user "brand-new-user" uploads file with content "uploaded content" to "fileToUpload.txt" using the WebDAV API
+    And the content of file "fileToUpload.txt" for user "brand-new-user" should be "uploaded content"
+
+
+  Scenario: should not be possible to disable encryption without decrypting encrypted files
```
Comment (phil-davis) [inline tests/acceptance/features/cliEncryption/encryption.feature:49]:
```suggestion
  Scenario: it should not be possible to disable encryption without decrypting encrypted files
```

File: tests/acceptance/features/cliEncryption/encryption.feature [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+    When the administrator invokes occ command "encryption:disable"
+    Then the command should have failed with exit code 1
+    And the command output should contain the text "The system still have encrypted files. Please decrypt them all before disabling encryption."
+
+  @skipOnOcV10 @issue-encryption-303
+  Scenario: should be possible to disable encryption after decrypting all encrypted files
```
Comment (phil-davis) [inline tests/acceptance/features/cliEncryption/encryption.feature:55]:
```suggestion
  Scenario: it should be possible to disable encryption after decrypting all encrypted files
```

**Review by phil-davis — COMMENTED** [review]
Please rebase when you make changes. This PR got a bit old, and lots has happened in core the last week.

**Comment by kiranparajuli589** [comment]
@phil-davis addressed your comments. please review

File: tests/acceptance/features/cliEncryption/encryption.feature [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+    When the administrator invokes occ command "encryption:status"
+    Then the command should have been successful
+    And the command output should contain the text "enabled: false"
+
+
+  Scenario: move encryption keys to a different folder
+    When the administrator invokes occ command "encryption:change-key-storage-root owncloud-keys"
+    Then the command should have been successful
+    When the administrator invokes occ command "encryption:show-key-storage-root"
+    And the command output should contain the text "Current key storage root:  owncloud-keys"
```
Comment (SwikritiT) [inline tests/acceptance/features/cliEncryption/encryption.feature:64]:
```suggestion
    Then the command output should contain the text "Current key storage root:  owncloud-keys"
```

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>cliEncryption-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/32771/122/1

**Review by SwikritiT — COMMENTED** [review]
LGTM

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>cliEncryption-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/32802/122/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>cliEncryption-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/32806/122/1

File: tests/acceptance/features/bootstrap/OccContext.php [hunk 11]
```diff
@@ -711,6 +721,31 @@ public function theAdministratorRunsUpgradeRoutinesOnLocalServerUsingTheOccComma
...
+		$this->theCommandShouldHaveBeenSuccessful();
+	}
+
+	/**
+	 * @When the administrator runs encryption decrypt all using the occ command
+	 *
+	 * @return void
+	 */
+	public function theAdministratorRunsEncryptionDecryptAllUsingTheOccCommand():void {
+		\system("echo 'root' | sudo -S www-data ./occ encryption:decrypt-all -c yes", $status);
```
Comment (kiranparajuli589) [inline tests/acceptance/features/bootstrap/OccContext.php:741]: what is the password for the drone root user?
Screenshot from 2021-10-01 15-44-52
https://drone.owncloud.com/owncloud/core/32806/122/13

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>cliEncryption-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/33195/123/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>cliEncryption-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/33221/4/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>cliEncryption-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/33232/4/1

**Review by phil-davis — APPROVED** [review]

**Pull request merged**